amplify/data/resource.ts

```diff
@@ -68,10 +69,20 @@ const schema = a
       .returns(a.ref("GenericFunctionResponse"))
       .authorization((allow) => [allow.guest(), allow.authenticated()])
       .handler(a.handler.function(AssignUsersToTeams)),
+    GenerateTeamCode: a
+      .mutation()
+      .arguments({
+        teamName: a.string().required(),
+        addCallerToTeam: a.boolean().required(),
+      })
+      .returns(a.ref("GenericFunctionResponse"))
+      .authorization((allow) => [allow.guest(), allow.authenticated()])
+      .handler(a.handler.function(GenerateTeamCode)),
```

Comment:
Last small thing, can we rename this lambda to CreateTeamWithCode now that it also creates the team? Just for clarity.